### ISSUE TITLE
[DatePicker] `aria-label` is missing on date picker day buttons after upgrading to ` v5.0.0-beta.7`

### ISSUE THREAD

**Issue by mschaefer-gresham** [mui/mui-x#6151]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

The button element for a day in the data picker does not have an aria-label after upgrading to react 18.

`<button class="MuiButtonBase-root MuiPickersDay-root Mui-selected MuiPickersDay-dayWithMargin css-gpd57x-MuiButtonBase-root-MuiPickersDay-root" tabindex="0" type="button" role="gridcell" aria-selected="true">15<span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"></span></button>`

### Expected behavior 🤔

I would expect there to be an aria-label as before:

`<button class="MuiButtonBase-root MuiPickersDay-root Mui-selected MuiPickersDay-dayWithMargin css-1i2xqvb" tabindex="0" type="button" aria-label="Sep 15, 2022">15<span class="MuiTouchRipple-root css-w0pj6f"></span></button>`

### Steps to reproduce 🕹

Steps:

1. Open date picker
2. Inspect any of the calendar day buttons.
3. See that there is no aria-label.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
System:
    OS: macOS 12.5.1
  Binaries:
    Node: 16.17.0 - ~/.nvm/versions/node/v16.17.0/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.15.0 - ~/.nvm/versions/node/v16.17.0/bin/npm
  Browsers:
    Chrome: 105.0.5195.125
    Edge: Not Found
    Firefox: 86.0.1
    Safari: 15.6.1
  npmPackages:
    @emotion/react: ^11.7.1 => 11.7.1 
    @emotion/styled: ^11.6.0 => 11.6.0 
    @mui/base:  5.0.0-alpha.86 
    @mui/envinfo: ^2.0.6 => 2.0.6 
    @mui/icons-material: ^5.8.4 => 5.8.4 
    @mui/lab: ^5.0.0-alpha.87 => 5.0.0-alpha.87 
    @mui/material: ^5.8.5 => 5.8.5 
    @mui/private-theming:  5.8.4 
    @mui/styled-engine:  5.8.0 
    @mui/styles: 5.8.4 => 5.8.4 
    @mui/system: ^5.8.5 => 5.8.5 
    @mui/types:  7.1.4 
    @mui/utils:  5.8.4 
    @mui/x-data-grid:  5.15.2 
    @mui/x-data-grid-premium: ^5.15.1 => 5.15.2 
    @mui/x-data-grid-pro:  5.15.2 
    @mui/x-date-pickers: ^5.0.1 => 5.0.1 
    @mui/x-date-pickers-pro: ^5.0.1 => 5.0.1 
    @mui/x-license-pro:  5.15.0 
    @types/react: ^18.0.17 => 18.0.19 
    react: ^18.2.0 => 18.2.0 
    react-dom: ^18.2.0 => 18.2.0 
    typescript: <4.5.5 => 4.5.4 

```
</details>

**Comment by LukasTy** [comment]
Hello @mschaefer-gresham 
What is the reason you are missing `aria-label` attribute on the day element?
We've recently done some [a11y improvement](https://github.com/mui/mui-x/pull/5809) on calendar/day picker.
The aria-label was explicitly removed to allow for correct screen reader behavior (tested on one of the industry standard tools - JAWS).
FYI, VoiceOver on Mac does not work well in this case as it seems lacking in ability to relate `column header` element and pronounce it's label when user is navigating the grid.

**Comment by mschaefer-gresham** [comment]
@LukasTy We were using it for testing purposes. But have found a workaround which we will just stay with if this was an expected change.

**Comment by lnmp4000** [comment]
Hi, @mschaefer-gresham  I was also using the aria-label to drive some test scripts. Can you tell me how you work round the removal of this?

**Comment by LukasTy** [comment]
Hello @lnmp4000 
You can take a look at the changes made in the linked PR.
If selecting a day in a range calendar is needed, we used [the following utility function](https://github.com/mui/mui-x/pull/5809/files#diff-87e0ff07902c0cb7c96c631ec1e5a3aecf63043e30613178f9e93cf4c1618f8cR37).
And for general pickers (with one calendar) [this worked fine](https://github.com/mui/mui-x/pull/5809/files#diff-e988434568a758195fcfeda4d50c0db1a80c1efd14079ffb34c6bda06d0d8954R102). Do keep in mind, that it might cause issues if you are using pickers with `showDaysOutsideCurrentMonth` prop.
